packages/core/src/components/button/buttons.tsx

```diff
@@ -50,65 +37,21 @@ export class AnchorButton extends AbstractButton<HTMLAnchorElement> {
     public static displayName = "Blueprint.AnchorButton";
 
     public render() {
-        const { children, href, onClick, loading, rightIconName, tabIndex = 0, text } = this.props;
-        const disabled = isButtonDisabled(this.props);
+        const { href, tabIndex = 0 } = this.props;
+        const props = this.getProps();
```

Comment:
Struggling to come up with a better method name. Maybe latch onto a different term like `getAttributes` / `getNativeElementAttributes` / `getNativeElementProps`? And I agree that we should prefer a different var name that doesn't suggest we're referring to props for the current-scope component.